### DISCUSSION TITLE
Default behaviour of closing the tooltip

### DISCUSSION THREAD

**Discussion by ZeCarlosCoutinho** [gilbarbara/react-joyride#768] [General]
When closing the tooltip, the event type generated is `step:after`. 
Why not something like `tooltip:close` (and the opposite to also exist: `tooltip:open`)? By looking at the names, the event `step:after` does not seem to be connected with the tooltip.

What is the reason to have this `step:after` associated with the action of closing the tooltip?
Maybe my knowledge of the events is not enough to understand this, but I also could not understand this subject better from what I read in the docs :(

**Comment by gilbarbara** [comment]
hey @ZeCarlosCoutinho 

I don't think I've understood your question, but the tooltip is just a part of the step lifecycle.

- step:before
- beacon
- tooltip
- step:after 

https://github.com/gilbarbara/react-joyride/blob/master/src/constants/events.js

**Reply by ZeCarlosCoutinho** [reply]
Well, this is a bit confusing, because then the event.js represent the steps lifecycle, but then there also exists a lifecycle.js file.
I still haven't understood when to use the values of the the `Events` or `Lifecycle`.

Anyway, from what you said, it seems that every action of the user triggers this cycle of step:before -> beacon -> tooltip -> step:after.

**Comment by gilbarbara** [comment]
The step lifecycle is described by the Lifecycle file while the Events file includes the tour and steps interactions.
